src/Theme/ThemeNegotiator.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\varbase_layout_builder\Theme\ThemeNegotiator
+ */
+namespace Drupal\varbase_layout_builder\Theme;
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Theme\AjaxBasePageNegotiator;
+use Drupal\Core\Access\CsrfTokenGenerator;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+
+class ThemeNegotiator extends AjaxBasePageNegotiator {
+
+
+  /**
+   * The CSRF token generator.
+   *
+   * @var \Drupal\Core\Access\CsrfTokenGenerator
+   */
+  protected $csrfGenerator;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * Constructs a new AjaxBasePageNegotiator.
+   *
+   * @param \Drupal\Core\Access\CsrfTokenGenerator $token_generator
+   *   The CSRF token generator.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request_stack
+   *   The request stack used to retrieve the current request.
+   */
+  public function __construct(CsrfTokenGenerator $token_generator, ConfigFactoryInterface $config_factory, RequestStack $request_stack) {
+    $this->csrfGenerator = $token_generator;
+    $this->configFactory = $config_factory;
+    $this->requestStack = $request_stack;
+  }
+  /**
+   * Whether this theme negotiator should be used to set the theme.
+   * @param RouteMatchInterface $route_match
+   * @return bool
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    return $this->getTheme($route_match) ? true : false;
```

Comment:
Our rule is:
- If we have the config set to show Claro theme **AND** the AJAX dialogue opens in a modal, then we need Claro.

Relying on the `$route_match` only, means that you're disregarding the modal. This means that if someones decides to disable Layout Builder Modal, then Claro will load in the off-canvas dialogue.

I think a more rigorous check is needed here to make sure you're switching to Claro theme only when it loads in Modal. You can utilize `$this->requestStack` for this. See how core does it: https://git.drupalcode.org/project/drupal/-/blob/9.0.x/core/lib/Drupal/Core/Theme/AjaxBasePageNegotiator.php#L68